modules/javafx.graphics/src/main/java/javafx/scene/Scene.java

```diff
@@ -2090,19 +2109,24 @@ null, new Point3D(
      *                                                                         *
      **************************************************************************/
 
-    private void windowForSceneChanged(Window oldWindow, Window newWindow) {
+    private void windowForSceneChanged(Window oldWindow, Window window) {
         if (oldWindow != null) {
+            oldWindow.showingProperty().removeListener(sceneWindowShowingListener);
             oldWindow.focusedProperty().removeListener(sceneWindowFocusedListener);
         }
 
-        if (newWindow != null) {
-            newWindow.focusedProperty().addListener(sceneWindowFocusedListener);
-            setWindowFocused(newWindow.isFocused());
+        if (window != null) {
+            window.showingProperty().addListener(sceneWindowShowingListener);
+            window.focusedProperty().addListener(sceneWindowFocusedListener);
+            setWindowFocused(window.isFocused());
         } else {
             setWindowFocused(false);
         }
+
+        checkCleanDirtyNodes();
     }
 
+    private final ChangeListener<Boolean> sceneWindowShowingListener = (p, o, n) -> {checkCleanDirtyNodes(); } ;
```

Comment:
Minor: should be no space before `;`.
You could also remove the curly braces.

Reply:
I've removed now the curly braces.